src/components/maintenance/LrMaintenanceSupportBundle.vue

```diff
@@ -0,0 +1,272 @@
+                  >
+                </template>
+              </v-text-field>
+            </template>
+            <v-date-picker
+              color="csmprimary"
+              v-model="timePeriod"
+              :max="new Date().toISOString().slice(0, 10)"
+              range
+            ></v-date-picker>
+          </v-menu>
+        </div>
+      </div>
+
+      <v-divider></v-divider>
+
+      <div class="form-element-container">
+        <div class="form-element">
+          <label for="">Component</label>
+          <SgtDropdown
+            placeholder="Component"
+            :dropdownOptions="['CSM', 'S3']"
+            v-model="selectedComponent"
+            @download="downloadBundle(data)"
+          />
+        </div>
+        <div class="form-element">
+          <label for="">Text</label>
+          <v-text-field
+            v-model="filterText"
+            placeholder="Text"
+            outlined
+            dense
+          ></v-text-field>
+        </div>
+      </div>
+      <v-divider></v-divider>
+    </div>
+    <v-btn
+      color="primary"
+      class="bundle-cta-btn"
+      :disabled="isDisableCTA"
+      elevation="0"
+      @click="generateSupportBundle"
+      >Generate</v-btn
+    >
+
+    <SgtDataTable
+      :headers="supportBundleConfig.supportBundleTable.headers"
+      :records="supportBundleData"
+      :searchConfig="supportBundleConfig.searchConfig"
+      @zoom="moreInfoHandler"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Watch } from "vue-property-decorator";
+import SgtDropdown from "@/lib/components/SgtDropdown/SgtDropdown.vue";
+import SgtDataTable from "@/lib/components/SgtDataTable/SgtDataTable.vue";
+import { Api } from "../../services/Api";
+import { lrMaintenanceSupportBundleConst } from "./LrMaintenance.constant";
+import SgtDialog from "@/lib/components/SgtDialog/SgtDialog.vue";
+import { SgtDialogModel } from "@/lib/components/SgtDialog/SgtDialog.model";
+import { create } from "vue-modal-dialogs";
+import { formatTime } from "@/utils/CommonUtilFunctions";
+import moment from "moment";
+@Component({
+  name: "LrMaintenanceSupportBundle",
+  components: { SgtDropdown, SgtDataTable, SgtDialog },
+})
+export default class LrMaintenanceSupportBundle extends Vue {
+  selectedNode: string = "";
+  selectedComponent: string = "";
+  filterText: string = "";
+  dates: string = "";
+  timePeriod: string[] = [];
+  timePeriodMenu: boolean = false;
+  supportBundleData = [];
+  nodeOptions = [];
+  supportBundleConfig = lrMaintenanceSupportBundleConst;
+  moreInfoModel = create<SgtDialogModel>(SgtDialog);
+
+  get isDisableCTA() {
+    let isDisable = true;
+    if (
+      this.selectedNode &&
+      this.selectedComponent &&
+      this.timePeriod.length === 2
+    ) {
+      isDisable = false;
+    }
+    return isDisable;
+  }
+
+  public dateRange = {
+    startDate: 0,
+    endDate: 0,
+  };
+
+  @Watch("timePeriod")
+  public printDates() {
```

Comment:
what is this function used for? to compare?

Reply:
When sending dates to the backend we need to send them in UNIX format. The date picker component stores date values in string format. So before sending the data, we need to make this conversion.

Reply:
Unix time conversion will be done only when making the API call.